src/main/java/io/github/syst3ms/skriptparser/lang/base/ExecutableExpression.java

```diff
@@ -0,0 +1,45 @@
+package io.github.syst3ms.skriptparser.lang.base;
+
+import io.github.syst3ms.skriptparser.lang.Effect;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.SelfRegistrable;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.registration.SkriptRegistration;
+
+import java.util.Arrays;
+
+/**
+ * A base class for syntax that can be used as {@link Expression} or {@link Effect}.
+ * An example of this could be:
+ * <ul>
+ *     {@code replace all {x} in {y} with {z}} <br>
+ *     {@code print "%replace all {x} in {y} with {z}%}
+ * </ul>
+ * Both syntaxes are valid and this enables productive and better coding practices.
+ * <br>
+ * The class automatically implements the {@link #execute(TriggerContext)} to just call
+ * {@link #getValues(TriggerContext)}, ignoring the results.
+ * This behavior can obviously be overridden.
+ * Finally, this implements {@link SelfRegistrable}, enabling an easy registration process.
+ */
+public abstract class ExecutableExpression<T>
+        extends Effect
+        implements Expression<T>, SelfRegistrable {
+
+    @Override
+    protected void execute(TriggerContext ctx) {
+        getValues(ctx);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void register(SkriptRegistration reg, Object... args) {
+        Class<T> type = (Class<T>) args[0];
+        boolean isSingle = (boolean) args[1];
```

Comment:
This have to be declared ? Can you make `reg.addExpression(getClass(), (Class<T>) args[0], (boolean) args[1], patterns);` instead ? (And make this two lines as comment to remember that is for type and isSingle)

Reply:
In fact we could, but it makes the code not readable after this. Here we knows that ``args[1]`` is a boolean and is the ``isSingle`` property. With your way, we don't know what each arguments is, so I'm not agree with this way.

Reply:
I'm ok with your opinion but you can also add a comment `//addExpression(class, type, isSingle, patterns);` . Use too much variables in java is not the best for ram performance (I learn that yesterday > avoid the maximum of variable). But if you don't want for better reading it's also ok, I understand too.

Reply:
@Syst3ms What do you think about it?

Reply:
> I'm ok with your opinion but you can also add a comment `//addExpression(class, type, isSingle, patterns);` . Use too much variables in java is not the best for ram performance (I learn that yesterday > avoid the maximum of variable). But if you don't want for better reading it's also ok, I understand too.

Note that we are using some variables twice.
Also, the impact of removing it is neglactable in comparison to the readability it offers. Not sure why this is a discussion. It's literally 3 lines of code. There are worse coding practices in this project and especially in the Skript project, which is one giant spaghetti bolognaise of Java.